docs/contributors/code/e2e/README.md

```diff
@@ -0,0 +1,131 @@
+# End-to-End Testing
+
+This living document serves to prescribe instructions and best practices for writing end-to-end (E2E) tests with Playwright in the Gutenberg project.
+
+> **Note**
+> See the dedicated guide if you're working with the previous Jest + Puppeteer framework. See the [migration guide](https://github.com/WordPress/gutenberg/tree/HEAD/docs/contributors/code/e2e/MIGRATION.md) if you're migrating tests from Jest + Puppeteer.
+
+## Running tests
+
+```bash
+# Run all available tests.
+npm run test:e2e:playwright
+
+# Run in headed mode.
+npm run test:e2e:playwright -- --headed
+
+# Run tests with specific browsers (`chromium`, `firefox`, or `webkit`).
+npm run test:e2e:playwright -- --project=webkit --project=firefox
+
+# Run a single test file.
+npm run test:e2e:playwright -- <path_to_test_file> # E.g., npm run test:e2e:playwright -- site-editor/title.spec.js
+
+# Debugging.
+npm run test:e2e:playwright -- --debug
+```
+
+If you're developing in Linux, it currently requires testing Webkit browsers in headed mode. If you don't want to or can't run it with the GUI (e.g. if you don't have a graphic interface), prepend the command with [`xvfb-run`](https://manpages.ubuntu.com/manpages/xenial/man1/xvfb-run.1.html) to run it in a virtual environment.
+
+```bash
+# Run all available tests.
+xvfb-run npm run test:e2e:playwright
+
+# Only run webkit tests.
+xvfb-run -- npm run test:e2e:playwright -- --project=webkit
+```
+
+## Best practices
+
+<details>
+	<summary><h3>Forbid `$`, use `locator` instead</h3></summary>
+
+In fact, any API that returns `ElementHandle` is [discouraged](https://playwright.dev/docs/api/class-page#page-query-selector). This includes `$`, `$$`, `$eval`, `$$eval`, etc. [`Locator`](https://playwright.dev/docs/api/class-locator) is a much better API and can be used with playwright's [assertions](https://playwright.dev/docs/api/class-locatorassertions). This also works great with Page Object Model since that locator is lazy and doesn't return a promise.
+</details>
```

Comment:
Observation: The "Best practices" individual sections aren't massive, and details/summary usage might make it less readable. What do you think?

Reply:
Yeah, it seems like GitHub will render large margins between these details tags. I'm fine either way, my main concern is that it might make the "Common pitfalls" section less discoverable.

Reply:
True! I considered that too. We'd have to make sure future revisions will reflect the TOC though. I wonder if there's already a linter or something similar in place so that we don't have to introduce it in this PR 🤔.

Reply:
I found the ToC example here https://developer.wordpress.org/block-editor/contributors/code/react-native/integration-test-guide/, but it's manually maintained.